### PR TITLE
 Scenclass

### PR DESCRIPTION
**Pull request by gidden**
Same idea as #26, but making a `Scenario` class for specific message work

### CODE REVIEW

File: tests/test_core.py [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+def test_add_spatial_single(test_mp):
+    scen = Scenario(test_mp, *msg_args, version='new')
+    data = {'country': 'Austria'}
+    scen.add_spatial_sets(data)
+
+    exp = ['World', 'Austria']
+    obs = scen.set('node')
+    npt.assert_array_equal(obs, exp)
+
+    exp = ['World', 'global', 'country']
```
Comment (gidden) [inline tests/test_core.py]: FYI @danielhuppmann, I've confirmed that the `add_spatial_sets` only adds `country` and `state` in these tests. `World` and `global` come in automatically on the `ixmp` side.

File: message_ix/core.py [hunk 2]
```diff
@@ -0,0 +1,128 @@
...
+        data = {'year': [2010, 2020]}
+        data = {'year': [2010, 2020], 'firstmodelyear': 2020}
+        """
+        if 'year' not in data:
+            raise ValueError('"year" must be in temporal sets')
+        horizon = data['year']
+        self.add_set("year", horizon)
+
+        first = data['firstmodelyear'] if 'firstmodelyear' in data else horizon[0]
+        self.add_set("cat_year", ["firstmodelyear", first])
```
Comment (danielhuppmann) [inline message_ix/core.py]: please change to `self.add_cat('year', 'firstmodelyear', first, is_unique=True)`

File: tests/test_core.py (hunk 1 above)
Comment (danielhuppmann) [reply tests/test_core.py]: indeed, that's a bug on the Java source side, see https://github.com/iiasa/ixmp_source/issues/120

File: tests/test_core.py [hunk 3]
```diff
@@ -0,0 +1,85 @@
...
+    obs = scen.set('lvl_spatial')
+    npt.assert_array_equal(obs, exp)
+
+    exp = [['country', 'Austria', 'World'], ['country', 'Germany', 'World']]
+    obs = scen.set('map_spatial_hierarchy')
+    npt.assert_array_equal(obs, exp)
+
+
+def test_add_spatial_heirarchy(test_mp):
```
Comment (danielhuppmann) [inline tests/test_core.py]: please fix typo to `hierarchy`

**Comment by gidden** [comment]
'Should be g2g @danielhuppmann

**Comment by gidden** [comment]
now g2g!

**Comment by danielhuppmann** [comment]
looks beautiful!

**Pull request merged**